### ISSUE TITLE
GCEInstanceService.create doesn't return created instance

### ISSUE THREAD

**Issue by machristie** [CloudVE/cloudbridge#47]
The following code, based on the tutorial, doesn't work because GCEInstanceService.create doesn't return the created instance

```
debian_img_id = 'debian-8-jessie-v20170523'
img = provider.compute.images.get(debian_img_id)
inst_type = sorted([t for t in provider.compute.instance_types.list()
                    if t.vcpus >= 2 and t.ram >= 4],
                   key=lambda x: x.vcpus*x.ram)[0]
inst = provider.compute.instances.create(
    name='cloudbridge-intro', image=img, instance_type=inst_type)
# Wait until ready
inst.wait_till_ready()  # This is a blocking call
# Show instance state
print(inst.state)
```

Fails with error
```
Traceback (most recent call last):
  File "test_gce.py", line 29, in <module>
    inst.wait_till_ready()  # This is a blocking call
AttributeError: 'NoneType' object has no attribute 'wait_till_ready'
```

I confirmed through the Google Cloud Engine dashboard that the instance was in fact created.

**Comment by afgane** [comment]
I've tried adding `return GCEInstance(self.provider, response)` to the bottom of `GCEInstanceService.create` method but am getting `HttpAccessTokenRefreshError: invalid_scope: Empty or missing scope not allowed.` when calling the `wait_till_ready` method (as well as a few others, such as `refresh`). Ping @baizhang

**Comment by baizhang** [comment]
There are two issues here.

First, the create method did not return the newly created instance. And `return GCEInstance(self.provider, response)` did not work, because `response` is a zone operation resource (instead of the instance resource). Commit 5bd1d05 fixes this problems by returning the newly created instance and also making this call block till the operation is complete.

The second issue is due to a change made in commit 45a0bf5, where GCE image now uses its selfLink as its id. And we will make it consistent across GCE resources to use selfLink as id. So the following slightly modified code should be able to create and return a new GCE instance.

```python
from cloudbridge.cloud.factory import CloudProviderFactory, ProviderList
provider = CloudProviderFactory().create_provider(ProviderList.GCE, {})
debian_img_name = 'debian-8-jessie-v20170523'
debian_images = provider.compute.images.find(debian_img_name)
img = debian_images[0]
# Alternatively
# img = provider.compute.images.get(
#     'https://www.googleapis.com/compute/v1/projects/debian-cloud/global/images/debian-8-jessie-v20170523')
inst_type = sorted([t for t in provider.compute.instance_types.list()
                    if t.vcpus >= 2 and t.ram >= 4],
                   key=lambda x: x.vcpus*x.ram)[0]
inst = provider.compute.instances.create(
    name='cloudbridge-intro', image=img, instance_type=inst_type)

```

Also add @mbookman here.

**Comment by mbookman** [comment]
I believe that this can be closed.
It was fixed by Bai's change: 5bd1d052408dfa4eb7d0654f5f35dd45e9891533.
@machristie @nuwang @afgane, please confirm.
